src/Swift.Bindings/src/Model/TypeDecl/ProtocolConformance.cs

```diff
@@ -7,9 +7,19 @@ namespace BindingsGeneration;
 /// <summary>
 /// Represents a protocol conformance.
 /// </summary>
-/// <param name="TargetType">The type spec of the target type of the conformance</param>
-/// <param name="ProtocolSpec">The protocol spec of the conformance</param>
-public record ProtocolConformance(
-    NamedTypeSpec TargetType,
-    NamedTypeSpec ProtocolSpec
+/// <param name="ConformingType">The type that conforms to the protocol</param>
+/// <param name="Protocol">The protocol that the type conforms to</param>
+public record TypeConformance(
+    SwiftTypeName ConformingType,
+    SwiftTypeName Protocol
+);
+
+/// <summary>
+/// Represents a generic parameter conformance.
+/// </summary>
+/// <param name="GenericParameter">The generic parameter</param>
+/// <param name="Protocol">The protocol that the generic parameter conforms to</param>
+public record GenericParameterConformance(
+    NamedTypeSpec GenericParameter,
```

Comment:
Could we encapsulate all information in the `SwiftTypeName` and use `NamedTypeSpec` in the demangler only?

Reply:
So here GenericParameter means something like `T`. I don't see that as `SwiftTypeName` (as it is not a "real" type but a placeholder). We could probably use raw strings here to not use `NamedTypeSpec` or introduce some thin string wrapper for type safety. 

Moving away from using TypeSpecs outside of demangler will not be trivial. Basically the problem is that we parse some method declaration and we see e.g. a bound generic type. We need some mechanism which will allow us to extract the open generic name and names of each of the type arguments. NamedTypeSpec has this functionality. I think we should discuss this further.

Reply:
Yes -- @matouskozak has an idea to utilize the Swift demangler which provides an AST.